mteb/tasks/Retrieval/eng/BrightRetrieval.py

```diff
@@ -0,0 +1,132 @@
+from collections import defaultdict
+
+import datasets
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks import AbsTaskRetrieval, MultilingualTask
+
+DOMAINS = [
+    "biology",
+    "earth_science",
+    "economics",
+    "psychology",
+    "robotics",
+    "stackoverflow",
+    "sustainable_living",
+    "pony",
+    # "leetcode",
+    # "aops",
+    # "theoremqa_theorems",
+    # "theoremqa_questions"
+]
+
+DOMAINS_langs = {domain: [domain] for domain in DOMAINS}
+
+
+EVAL_SPLITS = ["standard", "long"]
+
+
+class BrightRetrieval(MultilingualTask, AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="BrightRetrieval",
+        dataset={
+            "path": "xlangai/BRIGHT",
+            "revision": "a75a0eb",
+            # "name": "narrativeqa",
+        },
+        reference="https://huggingface.co/datasets/xlangai/BRIGHT",
+        description=("Bright retrieval dataset."),
+        type="Retrieval",
+        category="s2p",
+        eval_splits=EVAL_SPLITS,
+        eval_langs=DOMAINS_langs,
+        main_score="ndcg_at_10",
+        date=None,
+        form=None,
+        domains=None,
+        task_subtypes=["Article retrieval"],
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=[],
+        text_creation=None,
```

Comment:
These needs to be filled out. Especially domain is probably a good case here.

Reply:
```suggestion
        date=("2024-03-01", "2024-06-01"),
        form="written",
        domains=["Non-fiction"],
        task_subtypes=["Article retrieval"],
        license="CC-BY-4.0",
        socioeconomic_status="low",
        annotations_creators="derived",
        dialect=[],
        text_creation="found",
```